### PR TITLE
Ensure DefaultViewLocationCache.Set will update value if cache key already exists

### PR DESCRIPTION
**Pull request by ghisguth**
Fix for #3335.

Consequent call to DefaultViewLocationCache should update cached value.

### CODE REVIEW

**Comment by Kagamine** [comment]
Please sign the CLA: https://cla2.dotnetfoundation.org

**Comment by ghisguth** [comment]
I work in Microsoft, so that should not be a problem. But had to ask LCA anyway. Hope to get approval from them on Monday.

**Comment by Kagamine** [comment]
Wow!

**Comment by davidfowl** [comment]
@ghisguth You still have to sign the CLA

**Comment by Kagamine** [comment]
。。。。。。

File: test/Microsoft.AspNet.Mvc.Razor.Test/DefaultViewLocationCacheTest.cs [hunk 1]
```diff
@@ -90,6 +90,50 @@ public void InvokingGetAfterSet_ReturnsCachedItem(ViewLocationExpanderContext co
         }
 
         [Theory]
+        [MemberData(nameof(CacheEntryData))]
+        public void InvokingGetAfterSecondSet_ReturnsCachedItem(ViewLocationExpanderContext context)
```
Comment (Kagamine) [inline test/Microsoft.AspNet.Mvc.Razor.Test/DefaultViewLocationCacheTest.cs]: I think that use [Fact] is better here.

**Comment by pranavkm** [comment]
We're considering a design change for VIewLocationExpanderCache, so this type might go away. See https://github.com/aspnet/Mvc/issues/3337

File: test/Microsoft.AspNet.Mvc.Razor.Test/DefaultViewLocationCacheTest.cs (hunk 1 above)
Comment (ghisguth) [inline test/Microsoft.AspNet.Mvc.Razor.Test/DefaultViewLocationCacheTest.cs]: Updated

**Comment by ghisguth** [comment]
@Kagamine, @davidfowl Signed CLA. Updated test attribute.

@pranavkm feel free to merge or discard pull request based on what you decide in #3337.

**Comment by pranavkm** [comment]
@ghisguth, we've an ongoing PR for the redesign which removes the `IViewLocationExpanderCache` and its implementations. I'll close this PR since any changes here would eventually get discarded.

**Pull request closed without merging**